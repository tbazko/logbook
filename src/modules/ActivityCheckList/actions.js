import * as types from 'types'

export const toggleItemCheckbox = (id, createdAt) =>
  ({ type: types.ActivityCheckList.TOGGLE_ITEM_CHECKBOX, id, createdAt })

export const removeListItem = (id, activityType) =>
  ({ type: types.ActivityCheckList.REMOVE_LIST_ITEM, id, activityType })
export const removeAll = () => ({ type: types.ActivityCheckList.REMOVE_ALL })


export const checkAll = timestamp => ({ type: types.ActivityCheckList.CHECK_ALL, timestamp })
export const uncheckAll = timestamp => ({ type: types.ActivityCheckList.UNCHECK_ALL, timestamp })
export const setDefaultCheckboxValue = (timestamp, activityTypes) =>
  ({ type: types.ActivityCheckList.SET_DEFAULT_VALUES, timestamp, activityTypes })

export const setActiveCheckList = timestamp => ({
  type: types.ActivityCheckList.SET_ACTIVE_CHECKLIST,
  timestamp,
})

