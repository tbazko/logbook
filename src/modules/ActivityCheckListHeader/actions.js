import * as types from 'types'
import moment from 'moment'

export const showPrevDate = timestamp => ({
  type: types.ActivityCheckListHeader.SHOW_PREV_DATE,
  timestamp: moment(timestamp).add(-1, 'days').startOf('day').format(),
})

export const showNextDate = (timestamp, activityTypes) => ({
  type: types.ActivityCheckListHeader.SHOW_NEXT_DATE,
  timestamp: moment(timestamp).add(1, 'days').startOf('day').format(),
  activityTypes,
})
