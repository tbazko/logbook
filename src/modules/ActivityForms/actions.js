import uuid from 'react-native-uuid'
import moment from 'moment'
import * as types from 'types'

export const addItem = title => ({
  type: types.ActivityForm.ADD_ACTIVITY_TYPE,
  id: uuid.v1(),
  title,
  createdAt: moment().startOf('day').format(),
})

export const editItem = (id, payload) => ({
  type: types.ActivityForm.EDIT_ACTIVITY_TYPE,
  id,
  payload,
})

export const setViewMode = viewMode => ({
  type: types.ActivityCheckListHeader.SET_VIEW_MODE,
  viewMode,
})

export const addItemError = error => ({ type: types.ActivityForm.ADD_ACTIVITY_ERROR, error })
export const removeItemError = () => ({ type: types.ActivityForm.REMOVE_ACTIVITY_ERROR })
