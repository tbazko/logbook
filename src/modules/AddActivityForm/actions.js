import uuid from 'react-native-uuid'
import moment from 'moment'
import * as types from 'types'

export const addItem = title => ({
  type: types.AddActivityForm.ADD_ACTIVITY_TYPE,
  id: uuid.v1(),
  title,
  timestamp: moment().startOf('day').unix(),
})
export const addItemError = error => ({ type: types.AddActivityForm.ADD_ACTIVITY_ERROR, error })
