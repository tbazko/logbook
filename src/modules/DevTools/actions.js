import moment from 'moment'
import * as types from 'types'

export const resetStore = () => ({
  type: types.DevTools.REMOVE_ALL_DATA,
})

const todayTS = moment().startOf('day').format()
const activityTypes = {
  'Yoga-Id': {
    title: 'Yoga',
  },
  'Sport-Id': {
    title: 'Sport',
  },
  'Vega-Id': {
    title: 'Vega',
  },
  'Meat-Id': {
    title: 'Meat',
  },
  'Alco-Id': {
    title: 'Alco',
  },
}

export const insertData = () => ({
  type: types.DevTools.INSERT_DATA,
  payload: {
    activityTypes,
    activityLogs: getActivityLogs(),
    activeLogTimestamp: todayTS,
  },
})

export function getActivityLogs() {
  const logs = {}

  for (let i = 0; i < 60; i++) {
    logs[moment(todayTS).subtract(i, 'day').format()] = {
      'Sport-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Vega-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Meat-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Alco-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Yoga-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
    }
  }

  return logs
}
