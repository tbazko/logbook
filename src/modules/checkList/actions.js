import moment from 'moment'
import uuid from 'react-native-uuid'
import * as types from './constants'

export const addItem = (id, title) => ({
  type: types.ADD_ITEM,
  id,
  title,
})

export const toggleItemCheckbox = (id, timestamp) =>
  ({ type: types.TOGGLE_ITEM_CHECKBOX, id, timestamp });
export const addListItem = title => ((dispatch) => {
  const id = uuid.v4();
  dispatch(addItem(id, title));
  dispatch(toggleItemCheckbox(id, moment().startOf('day').unix()));
});
export const removeListItem = id => ({ type: types.REMOVE_LIST_ITEM, id });
export const removeAll = () => ({ type: types.REMOVE_ALL });


export const checkAll = timestamp => ({ type: types.CHECK_ALL, timestamp });
export const uncheckAll = timestamp => ({ type: types.UNCHECK_ALL, timestamp });
export const setDefaultCheckboxValue = timestamp => ({ type: types.SET_DEFAULT_ITEMS, timestamp });

export const setActiveCheckList = timestamp => ({ type: types.SET_ACTIVE_CHECKLIST, timestamp });
export const showPrevCheckList = timestamp => ((dispatch) => {
  const prevTimestamp = moment.unix(timestamp).add(-1, 'days').startOf('day').unix();
  dispatch(setDefaultCheckboxValue(prevTimestamp));
  dispatch(setActiveCheckList(prevTimestamp));
})
export const showNextCheckList = timestamp => ((dispatch) => {
  const nextTimestamp = moment.unix(timestamp).add(1, 'days').startOf('day').unix();
  dispatch(setActiveCheckList(nextTimestamp));
});
