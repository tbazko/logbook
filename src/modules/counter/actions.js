import * as types from './constants';


export const incrementCounter = () => ({ type: types.INCREMENT_COUNTER });
export const decrementCounter = () => ({ type: types.DECREMENT_COUNTER });
export const clearCounter = () => ({ type: types.RESET_COUNTER });
