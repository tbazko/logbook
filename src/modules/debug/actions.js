import * as types from 'types'
import data from 'storeMocks/activities.mock'

export const insertData = () => {
  console.log('tamara dispatching', data)
  return {
    type: types.Debug.INSERT_DATA,
    payload: data,
  }
}
