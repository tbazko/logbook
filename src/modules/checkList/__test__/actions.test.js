import mockStore from 'redux-mock-store';
import * as action from '../actions';

describe('CheckList', () => {
  let store;

  beforeEach(() => {
    store = mockStore();
  });

  it('should dispatch "ADD_ITEM" with title "Test" and id "1"', async () => {
    await store.dispatch(action.addItem('1', 'Test'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "ADD_ITEM", "TOGGLE_ITEM_CHECKBOX"', async () => {
    await store.dispatch(action.addListItem('Test'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "REMOVE_LIST_ITEM" with id "1"', async () => {
    await store.dispatch(action.removeListItem('1'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "REMOVE_ALL"', async () => {
    await store.dispatch(action.removeAll());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "TOGGLE_ITEM_CHECKBOX" with id "1" and timestamp "timestamp-1"', async () => {
    await store.dispatch(action.toggleItemCheckbox('1', 'timestamp-1'));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('should dispatch "CHECK_ALL" with timestamp "timestamp-2"', async () => {
    await store.dispatch(action.checkAll('timestamp-2'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "UNCHECK_ALL" with timestamp "timestamp-3"', async () => {
    await store.dispatch(action.uncheckAll('timestamp-3'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "SET_DEFAULT_ITEMS" with timestamp "timestamp-3"', async () => {
    await store.dispatch(action.setDefaultCheckboxValue('timestamp-3'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch "SET_ACTIVE_CHECKLIST" with timestamp "timestamp-3"', async () => {
    await store.dispatch(action.setActiveCheckList('timestamp-3'));
    expect(store.getActions()).toMatchSnapshot();
  });
});
