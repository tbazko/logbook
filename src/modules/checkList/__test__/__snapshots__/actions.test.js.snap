// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckList should dispatch "ADD_ITEM" with title "Test" and id "1" 1`] = `
Array [
  Object {
    "id": "1",
    "title": "Test",
    "type": "stats/checkList/ADD_ITEM",
  },
]
`;

exports[`CheckList should dispatch "ADD_ITEM", "TOGGLE_ITEM_CHECKBOX" 1`] = `
Array [
  Object {
    "id": "1",
    "title": "Test",
    "type": "stats/checkList/ADD_ITEM",
  },
  Object {
    "id": "1",
    "timestamp": 1519340400,
    "type": "stats/checkList/TOGGLE_ITEM_CHECKBOX",
  },
]
`;

exports[`CheckList should dispatch "CHECK_ALL" with timestamp "timestamp-2" 1`] = `
Array [
  Object {
    "timestamp": "timestamp-2",
    "type": "stats/checkList/CHECK_ALL",
  },
]
`;

exports[`CheckList should dispatch "REMOVE_ALL" 1`] = `
Array [
  Object {
    "type": "stats/checkList/REMOVE_ALL",
  },
]
`;

exports[`CheckList should dispatch "REMOVE_LIST_ITEM" with id "1" 1`] = `
Array [
  Object {
    "id": "1",
    "type": "stats/checkList/REMOVE_LIST_ITEM",
  },
]
`;

exports[`CheckList should dispatch "SET_ACTIVE_CHECKLIST" with timestamp "timestamp-3" 1`] = `
Array [
  Object {
    "timestamp": "timestamp-3",
    "type": "stats/checkList/SET_ACTIVE_CHECKLIST",
  },
]
`;

exports[`CheckList should dispatch "SET_DEFAULT_ITEMS" with timestamp "timestamp-3" 1`] = `
Array [
  Object {
    "timestamp": "timestamp-3",
    "type": "stats/checkList/SET_DEFAULT_ITEMS",
  },
]
`;

exports[`CheckList should dispatch "TOGGLE_ITEM_CHECKBOX" with id "1" and timestamp "timestamp-1" 1`] = `
Array [
  Object {
    "id": "1",
    "timestamp": "timestamp-1",
    "type": "stats/checkList/TOGGLE_ITEM_CHECKBOX",
  },
]
`;

exports[`CheckList should dispatch "UNCHECK_ALL" with timestamp "timestamp-3" 1`] = `
Array [
  Object {
    "timestamp": "timestamp-3",
    "type": "stats/checkList/UNCHECK_ALL",
  },
]
`;
